diff --git a/common/dialer/default.go b/common/dialer/default.go
index 6b2379f..6c6ead3 100644
--- a/common/dialer/default.go
+++ b/common/dialer/default.go
@@ -158,10 +158,15 @@ func NewDefault(ctx context.Context, options option.DialerOptions) (*DefaultDial
 		if keepInterval == 0 {
 			keepInterval = C.TCPKeepAliveInterval
 		}
+		keepCount := options.TCPKeepAliveCount
+		if keepCount < 0 {
+			keepCount = 0
+		}
 		dialer.KeepAliveConfig = net.KeepAliveConfig{
 			Enable:   true,
 			Idle:     keepIdle,
 			Interval: keepInterval,
+			Count:    keepCount,
 		}
 	}
 	var udpFragment bool
diff --git a/common/listener/listener_tcp.go b/common/listener/listener_tcp.go
index 2164ff8..ed27032 100644
--- a/common/listener/listener_tcp.go
+++ b/common/listener/listener_tcp.go
@@ -46,10 +46,15 @@ func (l *Listener) ListenTCP() (net.Listener, error) {
 		if keepInterval == 0 {
 			keepInterval = C.TCPKeepAliveInterval
 		}
+		keepCount := l.listenOptions.TCPKeepAliveCount
+		if keepCount < 0 {
+			keepCount = 0
+		}
 		listenConfig.KeepAliveConfig = net.KeepAliveConfig{
 			Enable:   true,
 			Idle:     keepIdle,
 			Interval: keepInterval,
+			Count:    keepCount,
 		}
 	}
 	if l.listenOptions.TCPMultiPath {
diff --git a/option/inbound.go b/option/inbound.go
index 42930ec..3526847 100644
--- a/option/inbound.go
+++ b/option/inbound.go
@@ -68,6 +68,7 @@ type ListenOptions struct {
 	DisableTCPKeepAlive  bool               `json:"disable_tcp_keep_alive,omitempty"`
 	TCPKeepAlive         badoption.Duration `json:"tcp_keep_alive,omitempty"`
 	TCPKeepAliveInterval badoption.Duration `json:"tcp_keep_alive_interval,omitempty"`
+	TCPKeepAliveCount    int               `json:"tcp_keep_alive_count,omitempty"`
 	TCPFastOpen          bool               `json:"tcp_fast_open,omitempty"`
 	TCPMultiPath         bool               `json:"tcp_multi_path,omitempty"`
 	UDPFragment          *bool              `json:"udp_fragment,omitempty"`
diff --git a/option/outbound.go b/option/outbound.go
index 2ed7cf2..39ea2fc 100644
--- a/option/outbound.go
+++ b/option/outbound.go
@@ -80,6 +80,7 @@ type DialerOptions struct {
 	DisableTCPKeepAlive  bool                              `json:"disable_tcp_keep_alive,omitempty"`
 	TCPKeepAlive         badoption.Duration                `json:"tcp_keep_alive,omitempty"`
 	TCPKeepAliveInterval badoption.Duration                `json:"tcp_keep_alive_interval,omitempty"`
+	TCPKeepAliveCount    int                               `json:"tcp_keep_alive_count,omitempty"`
 	UDPFragment          *bool                             `json:"udp_fragment,omitempty"`
 	UDPFragmentDefault   bool                              `json:"-"`
 	DomainResolver       *DomainResolveOptions             `json:"domain_resolver,omitempty"`
